core/src/main/java/org/elasticsearch/rest/BytesRestResponse.java

```diff
@@ -135,9 +139,27 @@ private static XContentBuilder build(RestChannel channel, RestStatus status, Exc
         }
 
         XContentBuilder builder = channel.newErrorBuilder().startObject();
+        builder.field(STATUS, status.getStatus());
         ElasticsearchException.generateFailureXContent(builder, params, e, channel.detailedErrorsEnabled());
-        builder.field("status", status.getStatus());
         builder.endObject();
         return builder;
     }
+
+    public static ElasticsearchException errorFromXContent(XContentParser parser) throws IOException {
+        XContentParser.Token token = parser.nextToken();
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);
+
+        ElasticsearchException exception = null;
+
+        String currentFieldName = parser.currentName();
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+            }
+            if (STATUS.equals(currentFieldName) == false) {
+                exception = ElasticsearchException.failureFromXContent(parser);
+            }
```

Comment:
ignoring the status here makes sense to me, but I wonder if we are 100% sure that the status returned by `ElasticsearchException#status` (which looks at cause etc.) will always be the same as what we got back (either as status code or within response body).